pywbem_mock/_instancewriteprovider.py

```diff
@@ -270,367 +209,280 @@ def __init__(self, cimrepository=None):
         """
         super(InstanceWriteProvider, self).__init__(cimrepository)
 
     ####################################################################
     #
-    #   CreateInstance server request
+    #   CreateInstance mock WBEM server responder
     #
     ####################################################################
 
-    def CreateInstance(self, namespace, NewInstance):
+    def CreateInstance(self, namespace, new_instance):
         """
-        Implements a WBEM server responder for
+        Implements a mock WBEM server responder for
         :meth:`pywbem.WBEMConnection.CreateInstance`.
 
-        Create a CIM instance in the local CIM repository of this class.
-
-        The input parameters must have been validated before this method
-        is called including:
-
-        1. Namespace is valid in CIM repository.
-        2. The ``NewInstance`` is a valid CIM Instance.
-        3. All key properties defined in the class  of ``NewInstance`` are
-           in ``NewInstance``. This method cannot add properties or
-           property values to ``NewInstance``.
-        4. There are no properties in ``NewInstance`` that are not in the
-           corresponding class.
-        5. NewInstance path must not exist in the CIM repository.
+        Create a new CIM instance in the CIM repository of the mock WBEM server.
+
+        Validation already performed by the provider dispatcher that calls
+        this provider method:
+        - The provider method is called only for the registered class and
+          namespace.
+        - The Python types of all input parameters to this provider method are
+          as specified below.
+        - The namespace exists in the CIM repository.
+        - The creation class of the new instance exists in the namespace
+          in the CIM repository.
+        - All properties specified in the new instance are exposed (i.e.
+          defined and inherited with any overrides resolved) by the creation
+          class in the CIM repository, and have the same type-related
+          attributes (i.e. type, is_array, embedded_object).
+
+        Validation that should be performed by this provider method:
+        - new_instance does not specify any properties that are not
+          allowed to be initialized by the client, depending on the model
+          implemented by the provider.
+        - new_instance specifies all key properties needed by the
+          provider, depending on the model implemented by the provider.
+          The CIM repository will reject any new instance that does not have
+          all key properties specified.
+        - The new instance does not exist in the CIM repository.
+          This validation needs to be done by the provider because the
+          determination of the key properties for the instance path may depend
+          on the model implemented by the provider.
+          The CIM repository will reject the creation of instances that already
```

Comment:
Maybe not here but I wonder if I clearly stated somewhere that "already exists" means that an instance with the same path exists.  i.e the key is the path

Reply:
DONE - Added some clarification here.